Blog post - Quarto on HPC guide

fercer opened the pull request:
This is a post for the Blog section of the website about setting up Quarto on a HPC environment for rendering and preview of markdown documents.

fercer committed:
Revised version of the guide

fercer commented:
There are many changes to files that I didn't touch due to re-rendering the website.
I wonder if that could affect the rendering of the whole website.

erickmartins commented:
had a quick look and, other than the theme CSS files (which I have no idea why have changed), everything looks ok to me!

psobolewskiPhD commented:
I guess this is the reason to use an action for the render/publish, makes PR diffs cleaner?

psobolewskiPhD commented:
I *think* you don't need quarto in every conda env when using it in VSCode? 
I think you can have a user-installed quarto and then use `ipython` or `ipykernel` to make the kernel in an env accessible using, e.g.
`ipython kernel install --name "my-env" --user`
At least this worked for me locally.

psobolewskiPhD commented on posts/2024-08-27-quarto-on-HPC/index.qmd:
```diff
@@ -0,0 +1,166 @@
+# Install Quarto on a remote server (HPC)
+
+In this post, two options are presented to install Quarto on a HPC server.
+The first option installs Quarto on your home directory and works fine when creating code-less documents.
```
```suggestion
The first option installs Quarto in your home directory and works fine when creating code-less documents.

```

fercer replied:
I blame English and its inconsistent use of `on` and `in`!!!

psobolewskiPhD commented on posts/2024-08-27-quarto-on-HPC/index.qmd:
```diff
@@ -0,0 +1,166 @@
+That will generate the rendered versions of your markdowns.
+For `html`, `revealjs`, or `websites` formats, it will start a local server and provide a link that you can paste in your browser to visualize the result.
+
+# Authoring Jupyter notebooks with Quarto
```
This applies to other forms with code cells, like presentations too!
Maybe this heading should be made more generic, like `Authoring documents with code cells`

fercer replied:
I like that more general heading!

fercer replied:
I went for `Authoring documents with (Python) code cells` since I haven't tested with R, or any other language

psobolewskiPhD reviewed (approved):
Just a small nitpick and a comment.

fercer committed:
Update posts/2024-08-27-quarto-on-HPC/index.qmd

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

psobolewskiPhD commented on posts/2024-08-27-quarto-on-HPC/index.qmd:
```diff
@@ -0,0 +1,167 @@
+
+## Option 1. Install Quarto on your home directory
+
+Download the Quarto source code for *Linux x86 Tarball* from their [official website](https://quarto.org/docs/download/tarball.html?version=1.5.56&idPrefix=download) into your home directory.
```
Oh one more thought: this link has the version number in it, so it could become stale with a new version. Maybe just link to the page itself?
```suggestion
Download the Quarto source code for *Linux x86 Tarball* from their [official website](https://quarto.org/docs/download/tarball.html) into your home directory.

```
https://quarto.org/docs/download/tarball.html

fercer replied:
I didn't notice that the link contained the specific version when I copied it since it links to the general tarball download page.
Changing the link to the generic tarball download webpage a suggested!

fercer committed:
Update posts/2024-08-27-quarto-on-HPC/index.qmd

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

fercer committed:
Updated post to add setup for multiple virtual envs

fercer commented:
I added most of your suggestions @psobolewskiPhD, please take a look if you have some time!

Additionally, I changed the ordering of the installation options.
Since now the single conda environment option is shorter (easier?), I made it Option 1.
And the installation in the home directory is now Option 2, because requires more commands.

psobolewskiPhD commented on posts/2024-08-28-quarto-on-HPC/index.qmd:
````diff
@@ -0,0 +1,222 @@
+
+```
+conda activate my-venv
+ipython kernel install --name "my-venv" --user
````
```suggestion
ipykernel kernel install --name "my-venv" --user
```
I think this is the more general way -- jupyter depends on ipykernel. iPython works too, but folks using notebooks might not use it and may not have it installed?

fercer replied:
I think `ipykernel` requires `ipython`. But as you mention, `ipykernel` is more general and even the `ipython` documentation page show how to add kernels by using `ipykernel`, so I'll change it.

psobolewskiPhD replied:
oh, huh I thought it was the other way around, but yea looks like a jupyter install ends up with both, so I guess it doesn't matter then which command.

psobolewskiPhD commented on posts/2024-08-28-quarto-on-HPC/index.qmd:
````diff
@@ -0,0 +1,222 @@
+ipython kernel install --name "my-venv" --user
+```
+
+::: {.callout-note}
+If you are using `Option 1` for installing Quarto in a `conda` environment, you won't need to specify the kernel since it will use the one available in that virtual environment.
+:::
````
Maybe this should be above the section, so those folks know they can skip it?

fercer replied:
Yeah, I'll move the note above the section, so people who followed Option 1 can skip it.

fercer committed:
Updated kernel selection for authoring code cells
